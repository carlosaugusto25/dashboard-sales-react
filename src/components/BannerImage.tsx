import styled from "styled-components";
import loginImg from "../assets/banner-login.jpg";

export const BannerImage = styled.div`
    background-image: url(${loginImg});
    background-repeat: no-repeat;
    background-size: cover;
    height: 100vh;
    width: 50vw;
`;