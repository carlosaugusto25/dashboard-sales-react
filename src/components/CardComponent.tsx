import styled from "styled-components";
import { pixelToRem } from "@/utils";

export const CardComponent = styled.div`
    background-color: ${({ theme }) => theme.card.background};
    border-radius: ${pixelToRem(8)};
    border: ${pixelToRem(1)} solid ${({ theme }) => theme.card.border};
    box-sizing: border-box;
    padding: ${pixelToRem(24)};
    width: 100%;

    &.alert {
        background-color: ${({ theme }) => theme.card.alert};
        border-color: ${({ theme }) => theme.card.alert};
    }

    &.success {
        background-color: ${({ theme }) => theme.card.success};
        border-color: ${({ theme }) => theme.card.success};
    }

    &.warning {
        background-color: ${({ theme }) => theme.card.warning};
        border-color: ${({ theme }) => theme.card.warning};
    }

    a {
        text-decoration: none;
        color: inherit;
    }
`;