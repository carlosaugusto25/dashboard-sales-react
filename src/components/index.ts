export { AvatarsList } from './AvatarList';
export { FormComponent } from './FormComponent';
export { Header } from './Header';
export * from './BannerImage';
export * from './CardComponent';
export * from './Logo';
export * from './StyledButton';
export * from './StyledInput';
export * from './Typographies';