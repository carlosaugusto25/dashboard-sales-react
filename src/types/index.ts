export * from "./appThemeContext";
export * from "./avatarsList";
export * from "./customChart";
export * from "./customTable";
export * from "./decoderJwt";
export * from "./formComponent";
export * from "./highlightsData";
export * from "./leadsData";
export * from "./loginData";
export * from "./newsData";
export * from "./profileData";
export * from "./starsData";
export * from "./theme";
export * from "./typographies";
export * from "./validationPassword";
