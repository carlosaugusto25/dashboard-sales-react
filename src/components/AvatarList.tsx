import { AvatarsListProps } from "@/types";
import { Avatar, Box } from "@mui/material";
import { pixelToRem } from "@/utils";
import { StyledH2, StyledSpan } from "./Typographies";

export function AvatarsList(props: AvatarsListProps) {
  return (
    <>
      {props.listData.map((item, index) => (
        <Box
          sx={{
            alignItems: "center",
            display: "flex",
            padding: `${pixelToRem(12)} 0`,
            key: index,
          }}
        >
          <Box>
            <Avatar
              alt={item.name}
              src={item.avatar}
              sx={{
                width: pixelToRem(48),
                height: pixelToRem(48),
                marginRight: pixelToRem(16),
              }}
            />
          </Box>
          <Box>
            <StyledH2>{item.name}</StyledH2>
            <StyledSpan>{item.subtitle}</StyledSpan>
          </Box>
        </Box>
      ))}
    </>
  );
}
