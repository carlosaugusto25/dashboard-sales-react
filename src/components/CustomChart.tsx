import { useTheme } from "styled-components";
import { Chart as ChartJS, CategoryScale, LinearScale, BarElement, PointElement, LineElement, Title, Tooltip, Legend } from 'chart.js';
import { Bar, Line } from 'react-chartjs-2';
import { CustomChartProps } from "@/types";

ChartJS.register(CategoryScale, LinearScale, BarElement, PointElement, LineElement, Title, Tooltip, Legend);

export function CustomChart({ labels, data, type }: CustomChartProps) {

    const theme = useTheme();
    const options = {
        responsive: true,
        scaleShowVerticalLines: false,
        scales: {
            x: {
                border: {
                    display: false
                },
                grid: {
                    display: false
                },
                ticks: {
                    color: theme.typographies.subtitle
                }
            },
            y: {
                border: {
                    display: false
                },
                grid: {
                    color: theme.appDefaultStroke
                },
                ticks: {
                    color: theme.typographies.subtitle
                }
            }
        },
        plugins: {
            legend: {
                display: false
            }
        }
    }
    const chatData = {
        labels,
        datasets: [
            {
                data,
                borderColor: 'rgb(12,112,242)',
                backgroundColor: 'rgba(12,112,242,1)',
            }
        ]
    }

    return type === 'bar' ? (<Bar options={options} data={chatData} />) : (<Line options={options} data={chatData} />)
}