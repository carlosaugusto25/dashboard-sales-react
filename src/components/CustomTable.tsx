import styled from "styled-components";
import { pixelToRem } from "@/utils";
import { CustomTableProps } from "@/types";

const TableWrapper = styled.div`
  overflow-x: auto;
  width: 100%;

  table {
    width: 100%;
    border-collapse: collapse;

    .ellipsis {
      display: block;
      white-space: nowrap;
      text-overflow: ellipsis;
      overflow: hidden;
    }

    .ellipsis-sm {
      width: ${pixelToRem(300)};
    }

    .ellipsis-xs {
      width: ${pixelToRem(150)};
    }

    th,
    td {
      text-align: left;
      height: ${pixelToRem(48)};
      padding: 0 ${pixelToRem(8)} 0 0;
      &:last-child {
        text-align: right;
        padding: 0;
      }
    }

    th {
      color: ${({ theme }) => theme.typographies.subtitle};
      font-weight: 600;
    }

    tr {
      border-bottom: ${pixelToRem(1)} solid
        ${({ theme }) => theme.appDefaultStroke};
      &:last-child {
        border-bottom: none;
      }
    }
  }
`;

export function CustomTable({ headers, rows }: CustomTableProps) {
  return (
    <TableWrapper>
      <table>
        <thead>
          <tr>
            {headers.map((header, index) => (
              <th key={index}>{header}</th>
            ))}
          </tr>
        </thead>
        <tbody>
          {rows.map((row, index) => (
            <tr key={index}>
              {row.map((data, indx) => (
                <td key={indx}>{data}</td>
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </TableWrapper>
  );
}
