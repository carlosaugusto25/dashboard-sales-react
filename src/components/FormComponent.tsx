import { StyledButton, StyledInput } from "@/components";
import { FormComponentProps } from "@/types";
import styled from "styled-components";
import { pixelToRem } from "@/utils";

export const StyledForm = styled.form`
  display: flex;
  flex-direction: column;
  row-gap: ${pixelToRem(16)};
`;

export function FormComponent(props: FormComponentProps) {
  const { inputs, buttons, messages } = props;
  return (
    <StyledForm>
      {inputs.map((input, index) => (
        <StyledInput key={index} {...input} />
      ))}
      {buttons.map((button, index) => (
        <StyledButton key={index} {...button} />
      ))}
      {messages && (
        <div style={{ color: messages.type === "success" ? "green" : "red" }}>
          {messages.msg}
        </div>
      )}
    </StyledForm>
  );
}
