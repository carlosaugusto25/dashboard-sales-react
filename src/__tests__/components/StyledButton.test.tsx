import { StyledButton } from "@/components";
import { render } from "@testing-library/react";
import { Theme } from "@/types";
import { ThemeProvider } from "styled-components";
import { themeList } from "@/resources/themeList";

describe('StyledButton', () => {
    const renderComponent = (theme: Theme, className?: string, props = {}) =>
        render(
            <ThemeProvider theme={theme}>
                <StyledButton className={className} {...props} />
            </ThemeProvider>
        )

    themeList.forEach(({ name, theme }) => {
        describe(`${name}`, () => {
            it('should match the snapshot without alert class', () => {
                const { asFragment } = renderComponent(theme, 'alert')
                expect(asFragment()).toMatchSnapshot()
            })
            it('should match the snapshot without primary class', () => {
                const { asFragment } = renderComponent(theme, 'primary')
                expect(asFragment()).toMatchSnapshot()
            })
            it('should match the snapshot without borderless-alert class', () => {
                const { asFragment } = renderComponent(theme, 'borderless-alert')
                expect(asFragment()).toMatchSnapshot()
            })
            it('should match the snapshot without borderless-primary class', () => {
                const { asFragment } = renderComponent(theme, 'borderless-primary')
                expect(asFragment()).toMatchSnapshot()
            })
            it('should match the snapshot without disabled status', () => {
                const { asFragment } = renderComponent(theme, 'primary', { disabled: true })
                expect(asFragment()).toMatchSnapshot()
            })
        })
    })
})