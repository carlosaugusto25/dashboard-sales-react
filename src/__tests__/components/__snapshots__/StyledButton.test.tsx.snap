// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton Dark Theme should match the snapshot without alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  font-size: 1rem;
  font-weight: 700;
  height: 3.125rem;
  padding: 0 1rem;
  width: 100%;
  transition: background-color 0.3s;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0.borderless-primary {
  background-color: none;
  color: #fff;
  height: 0;
  padding: 0;
}

.c0.borderless-primary:hover {
  opacity: .7;
}

.c0:disabled {
  background-color: #313649;
  color: #6d7b8e;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
  color: #6d7b8e;
}

<button
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Dark Theme should match the snapshot without borderless-alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  font-size: 1rem;
  font-weight: 700;
  height: 3.125rem;
  padding: 0 1rem;
  width: 100%;
  transition: background-color 0.3s;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0.borderless-primary {
  background-color: none;
  color: #fff;
  height: 0;
  padding: 0;
}

.c0.borderless-primary:hover {
  opacity: .7;
}

.c0:disabled {
  background-color: #313649;
  color: #6d7b8e;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
  color: #6d7b8e;
}

<button
    class="c0 borderless-alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Dark Theme should match the snapshot without borderless-primary class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  font-size: 1rem;
  font-weight: 700;
  height: 3.125rem;
  padding: 0 1rem;
  width: 100%;
  transition: background-color 0.3s;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0.borderless-primary {
  background-color: none;
  color: #fff;
  height: 0;
  padding: 0;
}

.c0.borderless-primary:hover {
  opacity: .7;
}

.c0:disabled {
  background-color: #313649;
  color: #6d7b8e;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
  color: #6d7b8e;
}

<button
    class="c0 borderless-primary"
  />
</DocumentFragment>
`;

exports[`StyledButton Dark Theme should match the snapshot without disabled status 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  font-size: 1rem;
  font-weight: 700;
  height: 3.125rem;
  padding: 0 1rem;
  width: 100%;
  transition: background-color 0.3s;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0.borderless-primary {
  background-color: none;
  color: #fff;
  height: 0;
  padding: 0;
}

.c0.borderless-primary:hover {
  opacity: .7;
}

.c0:disabled {
  background-color: #313649;
  color: #6d7b8e;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
  color: #6d7b8e;
}

<button
    class="c0 primary"
    disabled=""
  />
</DocumentFragment>
`;

exports[`StyledButton Dark Theme should match the snapshot without primary class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  font-size: 1rem;
  font-weight: 700;
  height: 3.125rem;
  padding: 0 1rem;
  width: 100%;
  transition: background-color 0.3s;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0.borderless-primary {
  background-color: none;
  color: #fff;
  height: 0;
  padding: 0;
}

.c0.borderless-primary:hover {
  opacity: .7;
}

.c0:disabled {
  background-color: #313649;
  color: #6d7b8e;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
  color: #6d7b8e;
}

<button
    class="c0 primary"
  />
</DocumentFragment>
`;

exports[`StyledButton Light Theme should match the snapshot without alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  font-size: 1rem;
  font-weight: 700;
  height: 3.125rem;
  padding: 0 1rem;
  width: 100%;
  transition: background-color 0.3s;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0.borderless-primary {
  background-color: none;
  color: #0c70f2;
  height: 0;
  padding: 0;
}

.c0.borderless-primary:hover {
  opacity: .7;
}

.c0:disabled {
  background-color: #ccc;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #ccc;
  color: #666;
}

<button
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Light Theme should match the snapshot without borderless-alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  font-size: 1rem;
  font-weight: 700;
  height: 3.125rem;
  padding: 0 1rem;
  width: 100%;
  transition: background-color 0.3s;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0.borderless-primary {
  background-color: none;
  color: #0c70f2;
  height: 0;
  padding: 0;
}

.c0.borderless-primary:hover {
  opacity: .7;
}

.c0:disabled {
  background-color: #ccc;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #ccc;
  color: #666;
}

<button
    class="c0 borderless-alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Light Theme should match the snapshot without borderless-primary class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  font-size: 1rem;
  font-weight: 700;
  height: 3.125rem;
  padding: 0 1rem;
  width: 100%;
  transition: background-color 0.3s;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0.borderless-primary {
  background-color: none;
  color: #0c70f2;
  height: 0;
  padding: 0;
}

.c0.borderless-primary:hover {
  opacity: .7;
}

.c0:disabled {
  background-color: #ccc;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #ccc;
  color: #666;
}

<button
    class="c0 borderless-primary"
  />
</DocumentFragment>
`;

exports[`StyledButton Light Theme should match the snapshot without disabled status 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  font-size: 1rem;
  font-weight: 700;
  height: 3.125rem;
  padding: 0 1rem;
  width: 100%;
  transition: background-color 0.3s;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0.borderless-primary {
  background-color: none;
  color: #0c70f2;
  height: 0;
  padding: 0;
}

.c0.borderless-primary:hover {
  opacity: .7;
}

.c0:disabled {
  background-color: #ccc;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #ccc;
  color: #666;
}

<button
    class="c0 primary"
    disabled=""
  />
</DocumentFragment>
`;

exports[`StyledButton Light Theme should match the snapshot without primary class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  font-size: 1rem;
  font-weight: 700;
  height: 3.125rem;
  padding: 0 1rem;
  width: 100%;
  transition: background-color 0.3s;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0.borderless-primary {
  background-color: none;
  color: #0c70f2;
  height: 0;
  padding: 0;
}

.c0.borderless-primary:hover {
  opacity: .7;
}

.c0:disabled {
  background-color: #ccc;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #ccc;
  color: #666;
}

<button
    class="c0 primary"
  />
</DocumentFragment>
`;
